### PR TITLE
feat: openapi response schema

### LINKED ISSUE #930: Response schema missing in openapi docs
Screenshot 2024-08-17 at 18 58 39

### PR DESCRIPTION
**Pull request by VishnuSanal**
## Description

This PR fixes #930

## Summary

This PR adds response schema to openapi docs

## PR Checklist

Please ensure that:

- [x] The PR contains a descriptive title
- [x] The PR contains a descriptive summary of the changes
- [x] You build and test your changes before submitting a PR.
- [ ] You have added relevant documentation
- [ ] You have added relevant tests. We prefer integration tests wherever possible

## Pre-Commit Instructions:

- [x] Ensure that you have run the [pre-commit hooks](https://github.com/sansyrox/robyn#%EF%B8%8F-to-develop-locally) on your PR.

### CODE REVIEW

**Comment by VishnuSanal** [comment]
The schema will be defaulted to "any" for now; until typing is implemented. @sansyrox PTAL. I am not sure whether this was the expectation -- CMIIW.

File: robyn/openapi.py [hunk 1]
```diff
@@ -241,11 +241,7 @@ def get_path_obj(self, endpoint: str, summary: str, tags: List[str], query_param
             "summary": summary,
             "tags": tags,
             "parameters": openapi_parameter_object,
-            "responses": {
-                "200": {
-                    "description": "Successful Response",
-                }
-            },
+            "responses": {"200": {"description": "Successful Response", "content": {"application/json": {"schema": {}}}}},
```
Comment (sansyrox) [inline robyn/openapi.py:244]: @VishnuSanal ,
> Ensure that you have run the [pre-commit hooks](https://github.com/sansyrox/robyn#%EF%B8%8F-to-develop-locally) on your PR.

These were not run here

Comment (sansyrox) [inline robyn/openapi.py:244]: @VishnuSanal , we have typing at the moment. e.g. regular response vs json response. We need to distinguish between that on return type.

**Comment by sansyrox** [comment]
Screenshot 2024-08-19 at 19 48 57


@VishnuSanal , there is also an issue in rendering of the docstrings in the application.

The above is an expected response.


But what happens is the one below

Screenshot 2024-08-19 at 19 50 13

**Comment by sansyrox** [comment]
@VishnuSanal , could you fix it?

**Comment by VishnuSanal** [comment]
> @VishnuSanal , there is also an issue in rendering of the docstrings in the application.

what is the issue? the location where the description is shown? fastapi might be using a slightly different version of swagger; that must be the reason. should investigate. is this a major breaking feature? isn't it just about the aesthetics?

**Comment by sansyrox** [comment]
> what is the issue? the location where the description is shown? fastapi might be using a slightly different version of swagger; that must be the reason. should investigate. is this a major breaking feature? isn't it just about the aesthetics?

The issue is that Robyn's docstrings are not legible. We are appending them to the wrong location! This is may be due to a different version of swagger but feels more like wrong openapi.json generation.

Hence, a blocking issue, which needs to be fixed before merging

**Review by sansyrox — CHANGES_REQUESTED** [review]
Need to fix the openapi docs generation

**Comment by sansyrox** [comment]
The docstrings are being appended only to the `summary`. They should also be added to the `description` field.

**Comment by VishnuSanal** [comment]
@sansyrox done, PTAL. although unrelated, I have also removed openapi integration from future roadmap in the docs -- hope this is okay & need not create a new PR.

**Comment by VishnuSanal** [comment]
Todo: fix CI

**Pull request merged**